discovery-provider/src/tasks/index_network_peers.py

```diff
@@ -0,0 +1,174 @@
+        fetch_cnode_futures = {executor.submit(fetch_cnode_info, i, sp_factory_inst): i for i in ids_list}
+        for future in concurrent.futures.as_completed(fetch_cnode_futures):
+            single_cnode_fetch_op = fetch_cnode_futures[future]
+            try:
+                cn_endpoint_info = future.result()
+                # 1 index entry in the returned array from ServiceProviderFactory
+                eth_cn_endpoints[cn_endpoint_info[1]] = True
+            except Exception as exc:
+                logger.error(f"index_network_peers.py | fetch_cnode_futures {single_cnode_fetch_op} generated {exc}")
+    # Return dictionary with key = endpoint, formatted as { endpoint: True }
+    return eth_cn_endpoints
+
+# Determine the known set of distinct peers currently within a user replica set
+# This function differs from the above as we are not interacting with eth-contracts,
+#   instead we are pulling local db state and retrieving the relevant information
+def retrieve_peers_from_db(self):
+    db = update_network_peers.db
+    cnode_endpoints = {}
+    with db.scoped_session() as session:
+        db_cnode_endpts = (
+            session.query(
+                User.creator_node_endpoint).filter(
+                    User.creator_node_endpoint != None, User.is_current == True
+                ).distinct()
+        )
+        # Generate dictionary of unique creator node endpoints
+        for entry in db_cnode_endpts:
+            for cnode_user_set in entry:
+                cnode_entries = cnode_user_set.split(',')
+                for cnode_url in cnode_entries:
+                    if cnode_url == "''":
+                        continue
+                    cnode_endpoints[cnode_url] = True
+    return cnode_endpoints
+
+# calls GET identityservice/registered_creator_nodes to retrieve creator nodes currently registered on chain
+def fetch_cnode_endpoints_from_identity(self):
+    try:
+        identity_url = update_network_peers.shared_config['discprov']['identity_service_url']
+        identity_endpoint = urljoin(identity_url, 'registered_creator_nodes')
+
+        r = requests.get(identity_endpoint, timeout=3)
+        if r.status_code != 200:
+            raise Exception(f"Query to identity_endpoint failed with status code {r.status_code}")
+
+        registered_cnodes = r.json()
+        logger.info(f"Fetched registered creator nodes from chain via {identity_endpoint} - {registered_cnodes}")
+        return registered_cnodes
+    except Exception as e:
+        logger.error(f"Identity fetch failed {e}")
+        return []
+
+def refresh_cnodes_from_identity(self):
+    registered_cnodes = use_redis_cache(
+        'registered_cnodes_from_identity', 30, lambda: fetch_cnode_endpoints_from_identity(self)
+    )
+    return registered_cnodes
+
+
+# Function submitted to future in threadpool executor
+def connect_peer(endpoint, ipfs_client):
+    ipfs_swarm_address = get_ipfs_info_from_cnode_endpoint(endpoint, None)
```

Comment:
just curious how the error handling and timeout situation is in the get_ipfs_info_from_cnode_endpoint function and if the errors propagate in the connect_peers thread pool executor

Reply:
oh there's a try/except in the thread pool executor. resolving